main/inc/lib/hook/HookManagement.class.php

```diff
@@ -0,0 +1,516 @@
+            'id int UNSIGNED NOT NULL AUTO_INCREMENT, ' .
+            'hook_event_id int UNSIGNED NOT NULL, ' .
+            'hook_observer_id int UNSIGNED NOT NULL, ' .
+            'type tinyint NOT NULL, ' .
+            'hook_order int UNSIGNED NOT NULL, ' .
+            'enabled tinyint NOT NULL, ' .
+            'PRIMARY KEY PK_hook_management_hook_call (id) ' .
+            '); ';
+        Database::query($sql);
+    }
+
+    /**
+     * Initialize Database storing hooks (events, observers, calls)
+     * This should be called right after installDatabase method
+     * @return int
+     */
+    public function initDatabase()
+    {
+        // Search hook events
+        $hookEvents = array();
+        foreach (get_declared_classes() as $class) {
+            if (is_subclass_of($class, '\HookEvent')) {
+                $interfaces = class_implements($class);
+                $hookInterfaces = array();
+                foreach ($interfaces as $interface) {
+                    $hookInterface = (preg_filter('/Hook(.+)EventInterface/', '$1', $interface));
+                    if (!empty($hookInterface)) {
+                        $hookInterfaces[] = $hookInterface;
+                    }
+                }
+                $hookEvents[$class] = $hookInterfaces;
+            }
+        }
+        // Search hook observers
+        $hookObservers = array();
+        foreach (get_declared_classes() as $class) {
+            if (is_subclass_of($class, '\HookObserver')) {
+                $interfaces = class_implements($class);
+                $hookInterfaces = array();
+                foreach ($interfaces as $interface) {
+                    $hookInterface = (preg_filter('/Hook(.+)ObserverInterface/', '$1', $interface));
+                    if (!empty($hookInterface)) {
+                        $hookInterfaces[] =$hookInterface;
+                    }
+                }
+                $hookObservers[$class] = $hookInterfaces;
+            }
+        }
+        // Search hook calls
+        $hookCalls = array();
+        foreach ($hookEvents as $hookEvent => $eventInterfaces) {
+            if (!empty($eventInterfaces)) {
+                $order = 0;
+                foreach ($hookObservers as $hookObserver => $observerInterfaces) {
+                    if ($observerInterfaces === $eventInterfaces) {
+                        $order += 1;
+                        $hookCalls[] = array($hookEvent, $hookObserver, HOOK_TYPE_PRE, $order);
+                        $hookCalls[] = array($hookEvent, $hookObserver, HOOK_TYPE_POST, $order);
+                    }
+                }
+            }
+        }
+
+        // Insert hook events
+        foreach ($hookEvents as $hookEvent => $v) {
+            $attributes = array(
+                'class_name' => $hookEvent,
+                'description' => get_lang('HookDescription' . $hookEvent),
+            );
+            $id = Database::insert($this->tables[TABLE_PLUGIN_HOOK_EVENT], $attributes);
```

Comment:
This could be complicated to add it to db_main.sql, but basicly, this register all Hook classes found. Exist a method when installing a plugin with hooks, will check for Hook classes used and then register them.
Delete "init" method